File: app/src/main/java/com/example/my_hospital_appointments/MainPageDoctor.java
package com.example.my_hospital_appointments;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;

public class MainPageDoctor extends AppCompatActivity {

    String emailKey="";
    DatabaseReference myDoctorData;
    TextView helloUser,greetingUser;
    String doctorName;
    Button doctorAppointments,btnDoctorConsult,btnMedicalReports;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main_page_doctor);

        helloUser=(TextView)  findViewById(R.id.textViewHelloDr);
        greetingUser=(TextView)  findViewById(R.id.textViewHowArYouDr);

        Intent intent =getIntent();
        String myUsersEmail=intent.getExtras().getString("usersEmail");
        Toast.makeText(this, "DOC ID : "+myUsersEmail, Toast.LENGTH_SHORT).show();

        String [] parts= myUsersEmail.split("@");
        String emailID=parts[0];

        myDoctorData= FirebaseDatabase.getInstance().getReference("Doctors");

        getDoctorName(myUsersEmail);

        doctorAppointments=(Button) findViewById(R.id.buttonDoctorAppointments1);
        doctorAppointments.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
               // Intent myIntent=new Intent(MainPageDoctor.this,DoctorAppointments.class);
                Intent myIntent=new Intent(MainPageDoctor.this,Doctor_View_Appointments_NEW.class);
                myIntent.putExtra("docEmailId",emailID);
                startActivity(myIntent);
            }
        });

        btnDoctorConsult=(Button) findViewById(R.id.buttonDoctorConsultancy);
        btnDoctorConsult.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
              //  Intent myIntent=new Intent(MainPageDoctor.this,DoctorConsultation.class);
                Intent myIntent=new Intent(MainPageDoctor.this,Messages_From_Patients.class);
                myIntent.putExtra("doctorName",doctorName);
                startActivity(myIntent);
            }
        });

        btnMedicalReports=(Button) findViewById(R.id.buttonLabReportsDoctor);
        btnMedicalReports.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent myIntent=new Intent(MainPageDoctor.this,MedicalReport.class);
                myIntent.putExtra("usersEmail",myUsersEmail);
                startActivity(myIntent);
            }
        });

    }

    private void getDoctorName(String doctorEmail)
    {
        int Counter=doctorEmail.length();

        for(int a=0; a<Counter; a++)
        {
            if(doctorEmail.charAt(a)=='@')
            {
                break;
            }
            else
            {
                emailKey=emailKey+doctorEmail.charAt(a);

            }
        }

       // emailKey=emailKey.trim();

        myDoctorData.child(emailKey).get().addOnCompleteListener(new OnCompleteListener<DataSnapshot>() {
            @Override
            public void onComplete(@NonNull Task<DataSnapshot> task) {
                if(task.isSuccessful())
                {
                    if(task.getResult().exists())
                    {
                        DataSnapshot thisDataSnapshot=task.getResult();
                         doctorName=String.valueOf(thisDataSnapshot.child("userName").getValue());
                        if(!doctorName.isEmpty()) {
                            String Hello = "Hello Dr. "+doctorName;
                            String day="How Are You Doing Today?";
                            helloUser.setText(Hello);
                            greetingUser.setText(day);
                        }
                    }
                }
            }
        });



    }

    @Override
    protected void onResume() {
        super.onResume();
        Intent intent =getIntent();
        String myUsersEmail=intent.getExtras().getString("usersEmail");
        getDoctorName(myUsersEmail);
    }
}

File: app/src/main/java/com/example/my_hospital_appointments/MainPageDoctor.kt
package com.example.my_hospital_appointments

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase

class MainPageDoctor : AppCompatActivity() {
    var emailKey = ""
    var myDoctorData: DatabaseReference? = null
    var helloUser: TextView? = null
    var greetingUser: TextView? = null
    var doctorName: String? = null
    var doctorAppointments: Button? = null
    var btnDoctorConsult: Button? = null
    var btnMedicalReports: Button? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main_page_doctor)
        helloUser = findViewById<View>(R.id.textViewHelloDr) as TextView
        greetingUser = findViewById<View>(R.id.textViewHowArYouDr) as TextView
        val intent = intent
        val myUsersEmail = intent.extras!!.getString("usersEmail")
        Toast.makeText(this, "DOC ID : $myUsersEmail", Toast.LENGTH_SHORT).show()
        val parts =
            myUsersEmail!!.split("@".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        val emailID = parts[0]
        myDoctorData = FirebaseDatabase.getInstance().getReference("Doctors")
        getDoctorName(myUsersEmail)
        doctorAppointments = findViewById<View>(R.id.buttonDoctorAppointments1) as Button
        doctorAppointments!!.setOnClickListener { // Intent myIntent=new Intent(MainPageDoctor.this,DoctorAppointments.class);
            val myIntent = Intent(this@MainPageDoctor, Doctor_View_Appointments_NEW::class.java)
            myIntent.putExtra("docEmailId", emailID)
            startActivity(myIntent)
        }
        btnDoctorConsult = findViewById<View>(R.id.buttonDoctorConsultancy) as Button
        btnDoctorConsult!!.setOnClickListener { //  Intent myIntent=new Intent(MainPageDoctor.this,DoctorConsultation.class);
            val myIntent = Intent(this@MainPageDoctor, Messages_From_Patients::class.java)
            myIntent.putExtra("doctorName", doctorName)
            startActivity(myIntent)
        }
        btnMedicalReports = findViewById<View>(R.id.buttonLabReportsDoctor) as Button
        btnMedicalReports!!.setOnClickListener {
            val myIntent = Intent(this@MainPageDoctor, MedicalReport::class.java)
            myIntent.putExtra("usersEmail", myUsersEmail)
            startActivity(myIntent)
        }
    }

    private fun getDoctorName(doctorEmail: String?) {
        val Counter = doctorEmail!!.length
        for (a in 0 until Counter) {
            emailKey = if (doctorEmail[a] == '@') {
                break
            } else {
                emailKey + doctorEmail[a]
            }
        }

        // emailKey=emailKey.trim();
        myDoctorData!!.child(emailKey).get().addOnCompleteListener { task ->
            if (task.isSuccessful) {
                if (task.result.exists()) {
                    val thisDataSnapshot = task.result
                    doctorName = thisDataSnapshot.child("userName").value.toString()
                    if (!doctorName!!.isEmpty()) {
                        val Hello = "Hello Dr. $doctorName"
                        val day = "How Are You Doing Today?"
                        helloUser!!.text = Hello
                        greetingUser!!.text = day
                    }
                }
            }
        }
    }

    override fun onResume() {
        super.onResume()
        val intent = intent
        val myUsersEmail = intent.extras!!.getString("usersEmail")
        getDoctorName(myUsersEmail)
    }
}